src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/DateTime.cs

```diff
@@ -302,6 +318,57 @@ public override IEnumerable<TexlStrings.StringGetter[]> GetSignatures()
         }
     }
 
+    // DateValue(date_text:s, [languageCode:s]) : D
+    internal sealed class DateValueFunction_CO : DateTimeCustomObjectFunction
+    {
+        public override bool HasPreciseErrors => true;
+        public override bool SupportsParamCoercion => true;
+
+        public DateValueFunction_CO()
+            : base("DateValue", TexlStrings.AboutDateValue, DType.Date)
+        { }
+
+        public override IEnumerable<TexlStrings.StringGetter[]> GetSignatures()
+        {
+            yield return new[] { TexlStrings.DateValueArg1 };
+            yield return new[] { TexlStrings.DateValueArg1, TexlStrings.DateValueArg2 };
+        }
+    }
+
+    // TimeValue(time_text:s, [languageCode:s]) : T
+    internal sealed class TimeValueFunction_CO : DateTimeCustomObjectFunction
+    {
+        public override bool HasPreciseErrors => true;
+        public override bool SupportsParamCoercion => true;
+
+        public TimeValueFunction_CO()
+            : base("TimeValue", TexlStrings.AboutTimeValue, DType.Time)
+        { }
+
+        public override IEnumerable<TexlStrings.StringGetter[]> GetSignatures()
+        {
+            yield return new[] { TexlStrings.TimeValueArg1 };
+            yield return new[] { TexlStrings.TimeValueArg1, TexlStrings.TimeValueArg2 };
+        }
+    }
+
+    // DateTimeValue(time_text:s, [languageCode:s]) : d
+    internal sealed class DateTimeValueFunction_CO : DateTimeCustomObjectFunction
```

Comment:
might be possible to use a similar overload strategy for Text/Value to allow the routing to be done by the IR instead of at runtime

Reply:
Agreed. All functions which accept CustomObject should be their own special overloads and isolated from the rest of the language.

Reply:
This function has been removed